### ISSUE TITLE
Can't create new playlist with fresh Database

### ISSUE THREAD

**Issue by Faeb35** [airsonic-advanced/airsonic-advanced#277]
When running Airsonic Advanced, starting with a fresh database I'm getting the following error, when clicking on "Create new playlist".

Tested with version: 11.0.0-SNAPSHOT.20200501175424
Database: HSQLDB

```
2020-05-01 19:47:51.235 ERROR --- .WebSocketAnnotationMethodMessageHandler : Unhandled exception from message handler method

org.springframework.dao.DataIntegrityViolationException: PreparedStatementCallback; integrity constraint violation: NOT NULL check constraint; SYS_CT_10506 table: PLAYLIST column: DURATION_SECONDS; nested exception is java.sql.SQLIntegrityConstraintViolationException: integrity constraint violation: NOT NULL check constraint; SYS_CT_10506 table: PLAYLIST column: DURATION_SECONDS
	at org.springframework.jdbc.support.SQLExceptionSubclassTranslator.doTranslate(SQLExceptionSubclassTranslator.java:87) ~[spring-jdbc-5.2.5.RELEASE.jar:5.2.5.RELEASE]
	at org.springframework.jdbc.support.AbstractFallbackSQLExceptionTranslator.translate(AbstractFallbackSQLExceptionTranslator.java:72) ~[spring-jdbc-5.2.5.RELEASE.jar:5.2.5.RELEASE]
	at org.springframework.jdbc.support.AbstractFallbackSQLExceptionTranslator.translate(AbstractFallbackSQLExceptionTranslator.java:81) ~[spring-jdbc-5.2.5.RELEASE.jar:5.2.5.RELEASE]
	at org.springframework.jdbc.core.JdbcTemplate.translateException(JdbcTemplate.java:1443) ~[spring-jdbc-5.2.5.RELEASE.jar:5.2.5.RELEASE]
	at org.springframework.jdbc.core.JdbcTemplate.execute(JdbcTemplate.java:633) ~[spring-jdbc-5.2.5.RELEASE.jar:5.2.5.RELEASE]
	at org.springframework.jdbc.core.JdbcTemplate.update(JdbcTemplate.java:893) ~[spring-jdbc-5.2.5.RELEASE.jar:5.2.5.RELEASE]
	at org.springframework.jdbc.core.simple.AbstractJdbcInsert.executeInsertAndReturnKeyHolderInternal(AbstractJdbcInsert.java:433) ~[spring-jdbc-5.2.5.RELEASE.jar:5.2.5.RELEASE]
	at org.springframework.jdbc.core.simple.AbstractJdbcInsert.doExecuteAndReturnKeyHolder(AbstractJdbcInsert.java:394) ~[spring-jdbc-5.2.5.RELEASE.jar:5.2.5.RELEASE]
	at org.springframework.jdbc.core.simple.SimpleJdbcInsert.executeAndReturnKeyHolder(SimpleJdbcInsert.java:137) ~[spring-jdbc-5.2.5.RELEASE.jar:5.2.5.RELEASE]
	at org.airsonic.player.dao.AbstractDao.insert(AbstractDao.java:293) ~[classes/:na]
	at org.airsonic.player.dao.PlaylistDao.createPlaylist(PlaylistDao.java:92) ~[classes/:na]
	at org.airsonic.player.dao.PlaylistDao$$FastClassBySpringCGLIB$$e2635819.invoke(<generated>) ~[classes/:na]
	at org.springframework.cglib.proxy.MethodProxy.invoke(MethodProxy.java:218) ~[spring-core-5.2.5.RELEASE.jar:5.2.5.RELEASE]
	at org.springframework.aop.framework.CglibAopProxy$CglibMethodInvocation.invokeJoinpoint(CglibAopProxy.java:771) ~[spring-aop-5.2.5.RELEASE.jar:5.2.5.RELEASE]
	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:163) ~[spring-aop-5.2.5.RELEASE.jar:5.2.5.RELEASE]
	at org.springframework.aop.framework.CglibAopProxy$CglibMethodInvocation.proceed(CglibAopProxy.java:749) ~[spring-aop-5.2.5.RELEASE.jar:5.2.5.RELEASE]
	at org.springframework.dao.support.PersistenceExceptionTranslationInterceptor.invoke(PersistenceExceptionTranslationInterceptor.java:139) ~[spring-tx-5.2.5.RELEASE.jar:5.2.5.RELEASE]
	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:186) ~[spring-aop-5.2.5.RELEASE.jar:5.2.5.RELEASE]
	at org.springframework.aop.framework.CglibAopProxy$CglibMethodInvocation.proceed(CglibAopProxy.java:749) ~[spring-aop-5.2.5.RELEASE.jar:5.2.5.RELEASE]
	at org.springframework.aop.framework.CglibAopProxy$DynamicAdvisedInterceptor.intercept(CglibAopProxy.java:691) ~[spring-aop-5.2.5.RELEASE.jar:5.2.5.RELEASE]
	at org.airsonic.player.dao.PlaylistDao$$EnhancerBySpringCGLIB$$b1e49dcd.createPlaylist(<generated>) ~[classes/:na]
	at org.airsonic.player.service.PlaylistService.createPlaylist(PlaylistService.java:151) ~[classes/:na]
	at org.airsonic.player.ajax.PlaylistWSController.createEmptyPlaylist(PlaylistWSController.java:75) ~[classes/:na]
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[na:na]
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[na:na]
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[na:na]
	at java.base/java.lang.reflect.Method.invoke(Method.java:564) ~[na:na]
	at org.springframework.messaging.handler.invocation.InvocableHandlerMethod.doInvoke(InvocableHandlerMethod.java:171) ~[spring-messaging-5.2.5.RELEASE.jar:5.2.5.RELEASE]
	at org.springframework.messaging.handler.invocation.InvocableHandlerMethod.invoke(InvocableHandlerMethod.java:120) ~[spring-messaging-5.2.5.RELEASE.jar:5.2.5.RELEASE]
	at org.springframework.messaging.handler.invocation.AbstractMethodMessageHandler.handleMatch(AbstractMethodMessageHandler.java:565) ~[spring-messaging-5.2.5.RELEASE.jar:5.2.5.RELEASE]
	at org.springframework.messaging.simp.annotation.support.SimpAnnotationMethodMessageHandler.handleMatch(SimpAnnotationMethodMessageHandler.java:511) ~[spring-messaging-5.2.5.RELEASE.jar:5.2.5.RELEASE]
	at org.springframework.messaging.simp.annotation.support.SimpAnnotationMethodMessageHandler.handleMatch(SimpAnnotationMethodMessageHandler.java:94) ~[spring-messaging-5.2.5.RELEASE.jar:5.2.5.RELEASE]
	at org.springframework.messaging.handler.invocation.AbstractMethodMessageHandler.handleMessageInternal(AbstractMethodMessageHandler.java:520) ~[spring-messaging-5.2.5.RELEASE.jar:5.2.5.RELEASE]
	at org.springframework.messaging.handler.invocation.AbstractMethodMessageHandler.handleMessage(AbstractMethodMessageHandler.java:454) ~[spring-messaging-5.2.5.RELEASE.jar:5.2.5.RELEASE]
	at org.springframework.messaging.support.ExecutorSubscribableChannel$SendTask.run(ExecutorSubscribableChannel.java:144) ~[spring-messaging-5.2.5.RELEASE.jar:5.2.5.RELEASE]
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1130) ~[na:na]
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:630) ~[na:na]
	at java.base/java.lang.Thread.run(Thread.java:832) ~[na:na]
Caused by: java.sql.SQLIntegrityConstraintViolationException: integrity constraint violation: NOT NULL check constraint; SYS_CT_10506 table: PLAYLIST column: DURATION_SECONDS
	at org.hsqldb.jdbc.JDBCUtil.sqlException(Unknown Source) ~[hsqldb-2.5.0.jar:2.5.0]
	at org.hsqldb.jdbc.JDBCUtil.sqlException(Unknown Source) ~[hsqldb-2.5.0.jar:2.5.0]
	at org.hsqldb.jdbc.JDBCPreparedStatement.fetchResult(Unknown Source) ~[hsqldb-2.5.0.jar:2.5.0]
	at org.hsqldb.jdbc.JDBCPreparedStatement.executeUpdate(Unknown Source) ~[hsqldb-2.5.0.jar:2.5.0]
	at com.zaxxer.hikari.pool.ProxyPreparedStatement.executeUpdate(ProxyPreparedStatement.java:61) ~[HikariCP-3.4.2.jar:na]
	at com.zaxxer.hikari.pool.HikariProxyPreparedStatement.executeUpdate(HikariProxyPreparedStatement.java) ~[HikariCP-3.4.2.jar:na]
	at org.springframework.jdbc.core.JdbcTemplate.lambda$update$1(JdbcTemplate.java:894) ~[spring-jdbc-5.2.5.RELEASE.jar:5.2.5.RELEASE]
	at org.springframework.jdbc.core.JdbcTemplate.execute(JdbcTemplate.java:617) ~[spring-jdbc-5.2.5.RELEASE.jar:5.2.5.RELEASE]
	... 33 common frames omitted
Caused by: org.hsqldb.HsqlException: integrity constraint violation: NOT NULL check constraint; SYS_CT_10506 table: PLAYLIST column: DURATION_SECONDS
	at org.hsqldb.error.Error.error(Unknown Source) ~[hsqldb-2.5.0.jar:2.5.0]
	at org.hsqldb.Table.enforceRowConstraints(Unknown Source) ~[hsqldb-2.5.0.jar:2.5.0]
	at org.hsqldb.Table.generateAndCheckData(Unknown Source) ~[hsqldb-2.5.0.jar:2.5.0]
	at org.hsqldb.Table.insertSingleRow(Unknown Source) ~[hsqldb-2.5.0.jar:2.5.0]
	at org.hsqldb.StatementDML.insertRowSet(Unknown Source) ~[hsqldb-2.5.0.jar:2.5.0]
	at org.hsqldb.StatementInsert.getResult(Unknown Source) ~[hsqldb-2.5.0.jar:2.5.0]
	at org.hsqldb.StatementDMQL.execute(Unknown Source) ~[hsqldb-2.5.0.jar:2.5.0]
	at org.hsqldb.Session.executeCompiledStatement(Unknown Source) ~[hsqldb-2.5.0.jar:2.5.0]
	at org.hsqldb.Session.execute(Unknown Source) ~[hsqldb-2.5.0.jar:2.5.0]
	... 39 common frames omitted
```

Edit: Same issue when try to import a playlist
`Failed to import playlist "PreparedStatementCallback; integrity constraint violation: NOT NULL check constraint; SYS_CT_10506 table: PLAYLIST column: DURATION_SECONDS; nested exception is java.sql.SQLIntegrityConstraintViolationException: integrity constraint violation: NOT NULL check constraint; SYS_CT_10506 table: PLAYLIST column: DURATION_SECONDS".`

**Comment by randomnicode** [comment]
Unable to reproduce this bug

Sounds like somehow you missed a migration? Try the latest snapshot?

**Comment by Faeb35** [comment]
You're right, this is a custom built war. I built this using my fork on which I'm doing developments:
https://github.com/Faeb35/airsonic-advanced
I thought the printed version on the website would represent the version from which I forked, I just did a copy&paste from it.

So far I only changed very small code pieces, so I was not expecting that this would cause this issue.

After doing some more tests (and struggeling with #280), I can defentily say that the issue seems to be related to my development environment. Could be that I missed a migration there... Was not able to reproduce it with any releases wars on GitHub.

Sorry for all the confusion an wasting your time! In future I'll do some more test before raising an isse here :-(
